tests/Bootstrap/FileBootstrapProviderTest.php

```diff
@@ -75,7 +75,7 @@ public function testUnreadableFile()
         $file = $this->createTemporaryFile();
         chmod($file, 0222);
         $instance = new FileBootstrapProvider($file);
-        $this->expectException(RuntimeException::class);
+        $this->expectException(JsonException::class);
```

Comment:
@RikudouSage I couldn't check your comments as I'm struggling with this test that fails locally. I'm building this Dockerfile (`docker build . -t php-with-composer`) to reduce noise and make it reproducible:
```Dockerfile
FROM php:8.3.10-fpm

# Install system dependencies and clean cache
RUN apt-get update && apt-get install -y \
 git \
 curl \
 libpng-dev \
 libonig-dev \
 libxml2-dev \
 libcurl4-openssl-dev libssl-dev \ 
 zip \
 unzip

# Install PHP extensions
RUN docker-php-ext-install pdo_mysql mbstring exif pcntl bcmath gd

# Copy Composer from the official Composer image
COPY --from=composer:latest /usr/bin/composer /usr/bin/composer

COPY . /usr/src/unleash-client-php

WORKDIR /usr/src/unleash-client-php

RUN rm -f composer.lock

RUN composer install

CMD ["sh"]
```
And running the tests as: `docker run -it --rm --name my-running-script php-with-composer composer phpunit` and the test fails with the following error:

```shell
$ docker run -it --rm --name my-running-script php-with-composer composer phpunit
> phpunit
PHPUnit 9.6.21 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.10
Configuration: /usr/src/unleash-client-php/phpunit.xml

.........F.....................................................  63 / 233 ( 27%)
............................................................... 126 / 233 ( 54%)
............................................................... 189 / 233 ( 81%)
............................................                    233 / 233 (100%)

Time: 00:25.133, Memory: 20.00 MB

There was 1 failure:

1) Unleash\Client\Tests\Bootstrap\FileBootstrapProviderTest::testUnreadableFile
Failed asserting that exception of type "JsonException" matches expected exception "RuntimeException". Message was: "Syntax error" at
/usr/src/unleash-client-php/src/Bootstrap/FileBootstrapProvider.php:46
/usr/src/unleash-client-php/tests/Bootstrap/FileBootstrapProviderTest.php:79
.

FAILURES!
Tests: 233, Assertions: 895, Failures: 1.
Script phpunit handling the phpunit event returned with error code 1
```

Do you know what can it be?

Reply:
CI does prefer RuntimeException...

Reply:
I don't :/ But JsonException generally means that there was a syntax error in some JSON, any chance you're providing a non-JSON string somewhere where JSON is expected?

Reply:
Ah, figured it out, I assume you run the container under its default user, root? The test creates a file and makes it non-readable using chmod, but root has access to everything regardless of read/write permissions, thus the file is readable but because it's empty it fails with a JsonException.

If I run it as the current user, it works:

`docker run --rm -it -u $(id -u):$(id -g) php-with-composer vendor/bin/phpunit`